extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigClientConfig.java

```diff
@@ -129,6 +129,35 @@ public interface SpringCloudConfigClientConfig {
     @WithDefault("450")
     int ordinal();
 
+    /**
+     * Configuration for Config Server discovery.
+     */
+    DiscoveryConfig discovery();
+
+    interface DiscoveryConfig {
+        /**
+         * Enable discovery of the Spring Cloud Config Server
+         */
+        @WithDefault("false")
+        boolean enabled();
+
+        /**
+         * The service ID to use when discovering the Spring Cloud Config Server
+         */
+        String serviceId();
```

Comment:
Doesn't making this `String` instead of `Optional<String>` mean that an application without having set  `quarkus.spring-cloud-config.discovery.serviceId` will lead to a configuration error?